demos/sample_azure_iot_pnp/sample_azure_iot_pnp.h

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+#ifndef __SAMPLE_AZURE_IOT_PNP_H
+#define __SAMPLE_AZURE_IOT_PNP_H
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "azure_iot_hub_client_properties.h"
+
+// Header demo_config.h MUST define the Plug and Play Model ID
+// through the sampleazureiotMODEL_ID configuration.
+#ifndef sampleazureiotMODEL_ID
+    /**
+     * @brief The model id for this device
+     *
+     * https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/samples/Thermostat.json
+     *
+     * The model id is the JSON document (also called the Digital Twins Model Identifier or DTMI) which
+     * defines the capability of your device. The functionality of the device should match what is
+     * described in the corresponding DTMI. Should you choose to program your own PnP capable device,
+     * the functionality would need to match the DTMI and you would need to update the below 'model_id'.
+     * Please see the sample README for more information on this DTMI.
+     *
+     */
+    #define sampleazureiotMODEL_ID                                "dtmi:com:example:Thermostat;1"
+#endif
+
+extern AzureIoTHubClient_t xAzureIoTHubClient;
+
+/**
+ * @brief Provides the payload to be sent as telemetry to the Azure IoT Hub.
+ *
+ * @remark This function must be implemented by the specific sample.
+ *         `ulCreateTelemetry` is called periodically by the sample core task (the task created by `vStartDemoTask`). 
+ * 
+ * @param[out] pucTelemetryData    Pointer to uint8_t* that will contain the Telemetry payload.
+ * @param[in]  ulTelemetryDataSize Size of `pucTelemetryData`
+ * 
+ * @return uint32_t The number of bytes written in `pucTelemetryData`.
+ */
+uint32_t ulCreateTelemetry( uint8_t * pucTelemetryData,

```

Comment:
>ulCreateTelemetry [](http://example.com/codeflow?start=9&length=17)

how about returning size via argument and returning error vai return of this function?

Reply:
This applies to all the functions